### PR TITLE
feat: toggle side menu in small screen mode

### LINKED ISSUE #423: Can't close side-menu in small screen mode
**Describe the bug**
The side bar menu cannot be closed on a small screen format.

**To Reproduce**
1. Go to the application on a small screen device or resize your browser to a small screen format.
2. Open the side bar menu by clicking the menu icon.
3. Attempt to close the side bar menu by clicking the menu icon again.
4. Observe that the side bar menu does not close.

**Expected behavior**
When the side bar menu icon is clicked on a small screen format, the menu should open. Clicking the menu icon again should close the side bar menu.

**Screenshots**
Jul-31-2024 13-39-50

**Additional context**
This bug is only reproducible on small screen formats or when the browser is resized to mimic a small screen. The side bar menu functions as expected on larger screens.

### PR DESCRIPTION
**Pull request by muzanella11**
What's new:
- Add ability toggle side menu in small screen mode
- Improve toggle side menu behavior when page is changed or window resized

Closes #423 

Screenshoot:

https://github.com/user-attachments/assets/8df1d4d4-d4c4-45f6-a3b6-2e1400ca5c36

### CODE REVIEW

File: hosting/src/components/Sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -47,6 +36,30 @@ const Sidebar = ({sidebarOpen, setSidebarOpen}: SidebarProps) => {
     return () => document.removeEventListener("keydown", keyHandler);
   });
 
+  // close sidebar when page is changed or window resize
```
Comment (DennisAlund) [inline hosting/src/components/Sidebar/Sidebar.tsx:39]: Loving that detail 👏

**Review by DennisAlund — APPROVED** [review]

File: hosting/src/components/Sidebar/Sidebar.tsx [hunk 2]
```diff
@@ -49,6 +38,30 @@ const Sidebar = ({sidebarOpen, setSidebarOpen}: SidebarProps) => {
...
+
+    // Call handler right away so state gets updated with initial window size
+    handleResize();
+
+    // Clean up the event listener on component unmount
+    return () => window.removeEventListener("resize", handleResize);
+  }, []); // Empty array ensures that effect is only run on mount and unmount
+
   useEffect(() => {
     localStorage.setItem("sidebar-expanded", sidebarExpanded.toString());
```
Comment (detautama) [inline hosting/src/components/Sidebar/Sidebar.tsx:66]: What's the purpose of storing this item in local storage? I don't see where we're using that value retrieved from local storage.

Comment (muzanella11) [reply hosting/src/components/Sidebar/Sidebar.tsx:66]: It looks like it is old code and is no longer used. We can remove it from there

Comment (muzanella11) [reply hosting/src/components/Sidebar/Sidebar.tsx]: Done ✅ 
Thanks for the feedback @detautama

**Review by detautama — APPROVED** [review]

**Pull request merged**